Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Sliders/PinchSlider.cs

```diff
@@ -140,11 +188,27 @@ public void Start()
             {
                 throw new Exception($"Slider thumb on gameObject {gameObject.name} is not specified. Did you forget to set it?");
             }
+
+            if (startValue < minValue || startValue > maxValue)
+            {
+                startValue = ((maxValue - minValue) / 2) + minValue;
+            }
+            SliderValue = (startValue - minValue) / (maxValue - minValue);
+
             InitializeSliderThumb();
             OnValueUpdated.Invoke(new SliderEventData(sliderValue, sliderValue, null, this));
-        }
-
 
+            if (tickMarksAutoFill)
+            {
+                UpdateTickMarks();
+                StartCoroutine(LaunchUpdateTickMarks());
+            }
+        }
+        IEnumerator LaunchUpdateTickMarks()
+        {
+            yield return new WaitForSeconds(0.01f);
```

Comment:
Yes please! Sleeps are generalllly suuuper scary to have because they creep up on you (and generally gloss over something else underlying that is a real issue).

At times they are the only reasonable way of dealing with things, in which case they should be HEAVILY commented, otherwise what ends up happening is people end up copy-pasting the sleeps all over and lose context over what the original intention/reason for the sleep was.

But yeah in general sleeps scary!